sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -662,7 +668,8 @@ private[joins] class SortMergeJoinScanner(
     bufferedIter: RowIterator,
     inMemoryThreshold: Int,
     spillThreshold: Int,
-    eagerCleanupResources: () => Unit) {
+    eagerCleanupResources: () => Unit,
+    bufferFirstOnly: Boolean) {
```

Comment:
nit: `bufferFirstOnly` -> `matchedBufferFirstOnly`? And, please add `@param`, too.

Reply:
`bufferFirstOnly: Boolean = false` to avoid the unnecessary changes.

Reply:
Thanks, fixed.